### PR TITLE
2396 assign contacts with name

### PR DESCRIPTION
**Pull request by dankmitchell**
### Context
UCAS Contacts

### Changes proposed in this pull request
- Ensure all UCAS contacts have a name, telephone number and email

### Guidance to review
🚢 

### Checklist
- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by dankmitchell** [comment]
[Ensure all contacts have a "name" value](https://trello.com/c/X2vV1SqL/2396-ensure-all-contacts-have-a-name-value)

File: app/models/contact.rb [hunk 1]
```diff
@@ -21,6 +21,10 @@ class Contact < ApplicationRecord
 
   audited associated_with: :provider
 
+  validates :name, presence: true
+  validates :email, email: true, presence: true
+  validates :telephone, phone: { message: "^Enter a valid telephone number" }, allow_nil: true
```
Comment (defong) [inline app/models/contact.rb]: should message be in yml?

File: spec/models/contact_spec.rb [hunk 2]
```diff
@@ -45,5 +46,56 @@
       contacts.first.save
       expect(provider.reload.changed_at).to be_within(1.second).of Time.now.utc
     end
+
+
+    it { should validate_presence_of(:name) }
```
Comment (defong) [inline spec/models/contact_spec.rb]:
```suggestion
    it { should validate_presence_of(:name) }
    it { should validate_presence_of(:emai) }
```

File: app/models/contact.rb (hunk 1 above)
Comment (dankmitchell) [reply app/models/contact.rb]: Same as provider telephone number

**Comment by defong** [comment]
I think you forgot to up the schema number

File: db/migrate/20191023195941_assign_provider_name_to_contacts.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class AssignProviderNameToContacts < ActiveRecord::Migration[6.0]
+  def change
+    ucas_contacts_without_name = Contact.where(name: nil)
```
Comment (misaka) [inline db/migrate/20191023195941_assign_provider_name_to_contacts.rb]: We should probably add `.includes(:provider)` here for performance.

File: spec/models/contact_spec.rb [hunk 4]
```diff
@@ -45,5 +46,56 @@
...
+      end
+
+      it "validates email contains an @ symbol" do
+        contact.email = "bar"
+        contact.valid?
+
+        expect(contact.errors[:email]).to include("^Enter an email address in the correct format, like name@example.com")
+      end
+
+      it "Does not validate the email if it is not present"do
```
Comment (misaka) [inline spec/models/contact_spec.rb]: "Does not validate the email if it is not present"? Not sure that fits the test below, or am I missing something?

File: app/models/contact.rb (hunk 1 above)
Comment (misaka) [reply app/models/contact.rb]: I think @defong is right, we have a bunch of error messages already in locales, and this one probably belongs there.

https://github.com/DFE-Digital/manage-courses-backend/blob/413be4d4ada4407505ae6371614a929f80fbcf3a/config/locales/en.yml#L67-L68

File: db/migrate/20191023195941_assign_provider_name_to_contacts.rb (hunk 3 above)
Comment (dankmitchell) [reply db/migrate/20191023195941_assign_provider_name_to_contacts.rb]: Before: `1.0257s` 
After: `0.7702s`

💯

File: app/models/contact.rb (hunk 1 above)
Comment (dankmitchell) [reply app/models/contact.rb]: Ah yes, you are right.

File: spec/models/contact_spec.rb (hunk 4 above)
Comment (dankmitchell) [reply spec/models/contact_spec.rb]: Updated

File: db/migrate/20191023195941_assign_provider_name_to_contacts.rb (hunk 3 above)
Comment (misaka) [reply db/migrate/20191023195941_assign_provider_name_to_contacts.rb]: w00t! 😝 Ok, thought it would be a bit slower.

**Pull request merged**